irods/manager/data_object_manager.py

```diff
@@ -312,13 +313,23 @@ def put(
         updatables=(),
         **options
     ):
+        force = options.setdefault(
+            kw.FORCE_FLAG_KW, client_config.data_objects.force_put_by_default
+        )
+        if force or isinstance(force, str):
+            options[kw.FORCE_FLAG_KW] = ""
+        else:
+            del options[kw.FORCE_FLAG_KW]
```

Comment:
What happens if I have `force_put_by_default` set to `False` and I call `put()` with `FORCE_FLAG_KW` in the options? It looks like this will forcibly remove the `FORCE_FLAG_KW` from `options` if `force_put_by_default` is `False`, even if I want to explicitly use "force". In other words, it looks like the only way to use "force" is by configuring `force_put_by_default` to `True`.

Is that right?

Reply:
`setdefault` won't alter the value of the option passed in under the key of `kw.FORCE_FLAG_KW`;  it will return whatever was passed into the call under that key in `options`, unless that key was missing , in which case the default value of `client_config.data_objects.force_put_by_default` is used instead.

Reply:
Oooh, I see. Okay, please add a comment explaining what's going on there to help future readers.

Reply:
Hmm - I'm seeing another difficulty/ambiguity which must be resolved, so I'm thinking more changes to the code. 

Here's the "truth table" (and how things need to work) as I see it . (Key is T=True, F=False, np = not present, and "" can be considered as an equivalent for True in any input):
```
default_
setting  parameter result

T        T         ""
T        np        ""
T        F         np
F        T         ''
F        F/np      np
```

Reply:
code for  correctly resolving kw.FORCE_FLAG_KW appears right , just need better testing

Reply:
Resolved